### PR TITLE
fix(tls): add release_buffers mode

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: app/modules/_common_tls.lua [hunk 1]
```diff
@@ -254,7 +255,7 @@ function TLSSocket:_read(n)
         local i, o = self.out:pending()
         if i > 0 then
           net.Socket._write(self, self.out:read(), nil, function()
-            handshake()
+            timer.setImmediate(handshake)
```
Comment (creationix) [inline app/modules/_common_tls.lua]: Is the `_write` callback sync sometimes?  Or is there some other reason to need to defer handshake?

Comment (rphillips) [inline app/modules/_common_tls.lua]: _write shouldn't be sync, I can revert this change.

Comment (rphillips) [inline app/modules/_common_tls.lua]: fixed

**Comment by creationix** [comment]
+1

**Pull request merged**